Assets/Scripts/Player/PlayerController.cs

```diff
@@ -68,26 +82,52 @@ public class PlayerController : MonoBehaviour {
 
     public void TakeDamage(int damage)
     {
-
         currentHealth -= damage;
         healthBar.SetHealth(currentHealth);
 
-        if (currentHealth < 0)
-        {
-            currentHealth = 0;
-        }
+        if (currentHealth < 0) {  currentHealth = 0;  }
     }
 
     public void Heal(int heal)
     {
-
         currentHealth += heal;
         healthBar.SetHealth(currentHealth);
 
-        if (currentHealth > maxHealth)
+        if (currentHealth > maxHealth) {  currentHealth = maxHealth;  }
+    }
+
+    private void RangedAttack()

```

Comment:
It's not clear from the name of this method that it only works after a time threshold has been reached. I think it would be good to rename this method to `RangedAttackUpdate` (or something more clear). Alternatively, the code for `if (Time.time > nextFire)` and `nextFire = Time.time + fireRate;` could be moved outside of the `RangedAttack` method.
Similar changes could be made in `EnemyController.cs`.